File: 2-java-corba/1-hello-world/java/EchoServer.java
import org.omg.CORBA.ORB;
import org.omg.PortableServer.POAHelper;
import org.omg.PortableServer.POA;
import org.omg.CosNaming.NamingContextExt;
import org.omg.CosNaming.NamingContextExtHelper;
import org.omg.CosNaming.NameComponent;

public class EchoServer {
  public static void main(String args[]) {
    try {
      ORB orb = ORB.init(args, null); // create and initialize the ORB
      // get reference to rootpoa & activate the POAManager
      org.omg.CORBA.Object objRef = orb.resolve_initial_references("RootPOA");
      POA rootpoa = POAHelper.narrow(objRef);
      rootpoa.the_POAManager().activate();
      // get the naming service
      objRef = orb.resolve_initial_references("NameService");
      NamingContextExt ncRef = NamingContextExtHelper.narrow(objRef);
      // get object reference from the servant
      EchoImpl echoImpl = new EchoImpl();
      objRef = rootpoa.servant_to_reference(echoImpl);
      // convert the CORBA object reference into Echo reference
      Echo echoRef = EchoHelper.narrow(objRef);
      // bind the object reference in the naming service
      NameComponent path[ ] = ncRef.to_name("echo.echo"); // id.kind
      ncRef.rebind(path, echoRef);
      orb.run(); // start server...
    } catch (Exception e) {
      System.out.println("ERROR : " + e);
      e.printStackTrace(System.out);
    }
  }
}

File: 2-java-corba/1-hello-world/c++/EchoServer.cpp
#include <iostream>
#include <omniORB4/CORBA.h>
#include <omniORB4/Naming.hh>
#include "Echo.hh"

using namespace std;

class EchoImpl : public POA_Echo, public PortableServer::RefCountServantBase {
  public:
    virtual char* echoString(const char* msg) {
      cout << msg << endl;
      return CORBA::string_dup(msg);
    }
};

int main(int argc, char** argv) {
  CORBA::Object_var objRef;
  // create and initialize the ORB
  CORBA::ORB_var orb = CORBA::ORB_init(argc, argv);
  // get reference to rootpoa & activate the POAManager
  objRef = orb->resolve_initial_references("RootPOA");
  PortableServer::POA_var poaRef = PortableServer::POA::_narrow(objRef);
  poaRef->the_POAManager()->activate();
  // get the naming service
  objRef = orb->resolve_initial_references("NameService");
  CosNaming::NamingContext_var ncRef =
    CosNaming::NamingContext::_narrow(objRef);
  // instantiate the Echo CORBA object
  EchoImpl * echoImpl = new EchoImpl();
  Echo_var echoRef = echoImpl->_this();
  // bind the object reference in the naming service
  CosNaming::Name name;
  name.length(1);
  name[0].id
    = (const char*)"echo";
  name[0].kind = (const char*)"echo";
  ncRef->rebind(name, echoRef);
  // start server...
  orb->run();

  return 0;
}
